controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStream.java

```diff
@@ -812,6 +813,29 @@ public CompletableFuture<Map<UUID, ActiveTxnRecord>> getActiveTxns(OperationCont
         });
     }
 
+    @Override
+    public CompletableFuture<Map<UUID, TxnStatus>> listCompletedTxns(final OperationContext context) {
+        Preconditions.checkNotNull(context, "operation context cannot be null");
+        return getTxnInBatch(0, context);
+    }
+
+    private CompletableFuture<Map<UUID, TxnStatus>> getTxnInBatch(final Integer batch, final OperationContext context) {
+        Preconditions.checkNotNull(context, "operation context cannot be null");
+
+        Map<String, VersionedMetadata<CompletedTxnRecord>> result = new ConcurrentHashMap<>();
+        String tableName = getCompletedTransactionsBatchTableName(batch);
+        return storeHelper.expectingDataNotFound(storeHelper.getAllEntries(
+                tableName, CompletedTxnRecord::fromBytes, context.getRequestId()).collectRemaining(completedTxnMetadataMap -> {
+            result.put(completedTxnMetadataMap.getKey().replace(getCompletedTransactionKey(getScope(), getName(), ""), ""),
+                    completedTxnMetadataMap.getValue());
+            return true;
+        }).thenApply(v -> result.entrySet().stream().skip(result.size() > MAX_RECORD_BY_LIST_TXN_API ?
```

Comment:
Just using `skip` does not guarantee that we get the most recently added 500 transactions from the list of completed txns. We need to sort the CompletecTxnRecord list by `completeTime` and take the last 500 txns.

Reply:
Done as suggested.